mantidproject/mantid
# Reorganize tests of fit functions, category background

For the background functions, move the unit test cases that run the Fit algorithm to FitTest, and reorganized/extended the function tests.

**To test:**
- Code review
- Tests should pass on all platforms
- Ensure that:
  - The algorithm `Fit` is not used in the function unit tests, and the old cases that were using `Fit` are now in `FitTest`
  - No additional dependencies/headers/using are added in `FitTest`, and many headers/using are removed from the function unit tests.
- This also added some basic tests to evaluate functions, and check that wrong attribute/parameter values are handled gracefully. Otherwise some unit tests would have been left empty. It also adds a test on the category(ies) wherever it was missing. Also removes calls to AnalysisDataService::Instance().addOrReplace() / AnalysisDataService::Instance().remove() that are unnecessary and not what these tests are about.

Fixes #16479
This is one more step of #16267

_Does not need to be in the release notes._

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
